File: AtCoder Beginner Contest 193/B - Play Snuke.py
# https://atcoder.jp/contests/abc193/tasks/abc193_b

n = int(input())

apx_list = [[int(i) for i in input().split()] for j in range(n)]

# print(n)
# print(apx_list)

count = -1
temp = 0
for i in range(n):
    if(apx_list[i][2]-1-((apx_list[i][0]*60-30)//60) > 0):
        temp = apx_list[i][1]
        if(count == -1):
            count = temp
        elif(count > temp):
            count = temp
        else:
            pass
        # count += apx_list[i][1]
        # print(count)
    else:
        pass

print(count)

File: AtCoder Beginner Contest 193/B - Play Snuke.cpp
// https://atcoder.jp/contests/abc193/tasks/abc193_b

#include<bits/stdc++.h>
using namespace std;

int main(){
  int N;
  cin >> N;

  vector<double> A(N), P(N), X(N);
  for(int i = 0; i < N; i++){
    cin >> A[i] >> P[i] >> X[i];
  }
  
  int min = -1;
  double temp = 0;
  for(int i = 0; i < N; i++){
    //↓これは二重ループになる
    //while(0.5 < A[i]){
    //A[i] -= 1;
    //X[i] -= 1;
    if(X[i]-1-(((A[i])*60-30)/60) > 0){
      temp = P[i];
      if(min == -1){
        min = temp;
      }
      else if(temp < min){
        min = temp;
      }
      else{
        continue;
      }
    }
    else{
      continue;
    }
  }

  cout << min << endl;

  return 0;
}